scene/2d/physics_body_2d.cpp

```diff
@@ -475,36 +493,29 @@ void RigidBody2D::_body_state_changed(PhysicsDirectBodyState2D *p_state) {
 			}
 		}
 
-		_RigidBody2DInOut *toadd = (_RigidBody2DInOut *)alloca(p_state->get_contact_count() * sizeof(_RigidBody2DInOut));
+		RigidBody2D_AddAction *toadd = (RigidBody2D_AddAction *)alloca(p_state->get_contact_count() * sizeof(*toadd));
```

Comment:
```suggestion
		RigidBody2D_AddAction *toadd = (RigidBody2D_AddAction *)alloca(p_state->get_contact_count() * sizeof(RigidBody2D_AddAction));
```

Reply:
Just want to note that while I have implemented the requested change, it makes it easier to screw up this code in the future. The compiler knows the type that `toadd` points to; it will always allocate the correct amount of space. Hard-coding the type adds an opportunity for someone to forget to update their code (should the type of `toadd` ever change), and end up allocating the wrong size.

Reply:
It is more confusing IMO, and we don't do this anywhere else AFAIK, it should be the correct type to make it clear

Reply:
There is a light sprinkling of other instances of this method, through the renderer, the GDScript module, and some of the Windows platform code. It's pretty much everywhere in various thirdparty modules, because enough developers have been bitten in the ass by changing the type in one place, and forgetting to change it in the other place, that it is a well-ingrained habit for long-time C developers. C++ tends not to want it as often, because there are more powerful ways to manage memory. That said, this particular line is a lot more "C" than "C++", which is why we have the ugliness of having to repeat our type three times, in the first place. I suspect the core engine probably provides better ways of doing this allocation, but I don't know what they are.

Regardless, I have made the change you requested.

Reply:
Well, like, I said, I updated it as you suggested.

Reply:
I know? You were stating your points, and I mine, I'm not sure what you're trying to say. I just said that it was my opinion and not objective 🙂

Reply:
Oh, no, not at all. I shared your interpretation, that we were discussing possible approaches, and rationale for them. I just noticed that you hadn't marked the conversation as resolved, as you had done with the others, so I wasn't sure if you were waiting for some further action from my side.

Reply:
Only left it open to continue discussing 🙂 I'm sorry for the confusion